packages/react-native-audio-api/common/cpp/audioapi/core/utils/AudioNodeManager.cpp

```diff
@@ -55,17 +59,21 @@ void AudioNodeManager::settlePendingConnections() {
 }
 
 void AudioNodeManager::prepareNodesForDestruction() {
-  auto it = nodes_.begin();
-
-  while (it != nodes_.end()) {
-    if (it->use_count() == 1) {
-      assert(it->get()->inputNodes_.empty());
-      it->get()->cleanup();
-      it = nodes_.erase(it);
-    } else {
-      ++it;
+  if (auto lock = Locker::tryLock(nodeDeconstructor_->getLock())) {
```

Comment:
Instead of calling public method `getLock`, maybe let's pass a `std::function` that will be called with a lock?